es2-persistence/src/main/java/com/netflix/conductor/dao/es/index/ElasticSearchDAO.java

```diff
@@ -279,28 +332,41 @@ public void indexTask(Task task) {
 
 	@Override
 	public void addTaskExecutionLogs(List<TaskExecLog> taskExecLogs) {
-
 		if (taskExecLogs.isEmpty()) {
 			return;
 		}
+
 		try {
+			long startTime = Instant.now().toEpochMilli();
 			BulkRequestBuilder bulkRequestBuilder = elasticSearchClient.prepareBulk();
 			for (TaskExecLog taskExecLog : taskExecLogs) {
 				IndexRequest request = new IndexRequest(logIndexName, LOG_DOC_TYPE);
 				request.source(objectMapper.writeValueAsBytes(taskExecLog));
 				bulkRequestBuilder.add(request);
 			}
-			new RetryUtil<BulkResponse>().retryOnException(() -> bulkRequestBuilder.execute().actionGet(),
-					null, BulkResponse::hasFailures, RETRY_COUNT, "Indexing all execution logs into doc_type task", "addTaskExecutionLogs");
-		} catch (Throwable e) {
-			logger.error("Indexing failed {}", e.getMessage(), e);
+			new RetryUtil<BulkResponse>().retryOnException(
```

Comment:
Why not use `updateWithRetry` helper method?

Reply:
Will refactor in a subsequent PR.